Source/DataSources/networkLinkUpdateMgr.js

```diff
@@ -0,0 +1,91 @@
+export default function networkLinkUpdateMgr() {
+  var kmlSources = [];
+
+  return {
+    addLink: function (href, kml, dataSource) {
+      kmlSources[href.request.url] = {
+        kml: kml,
+        dataSource: dataSource,
+      };
+    },
+
+    processUpdate: function (updateNode) {
+      processUpdate(kmlSources, updateNode);
+    },
+  };
+}
+
+function processUpdate(kmlSources, updateNode) {
+  var source = kmlSources[getUpdateTargetHref(updateNode)];
+  if (!source) return;
+
+  processUpdateNode(source.kml, updateNode);
+  source.dataSource.load(source.kml).then(function () {
+    source.dataSource._changed.raiseEvent(source);
+  });
+}
+
+function getUpdateTargetHref(updateNode) {
+  return updateNode.getElementsByTagName("targetHref")[0].innerHTML;
+}
+
+function processUpdateNode(kmlRoot, updateNode) {
+  forEachChild(updateNode, function (child) {
+    switch (child.tagName) {
+      case "Change":
+        processChangeNodes(kmlRoot, child);
+        break;
+      case "Create":
+        processCreateNodes(kmlRoot, child);
+        break;
+      case "Delete":
+        processDeleteNode(kmlRoot, child);
+        break;
+    }
+  });
+}
+
+function processChangeNodes(root, changeNode) {
+  forEachChild(changeNode, function (child) {
+    processChangeNode(root, child);
+  });
+}
+
+function processChangeNode(root, newNode) {
+  var oldNode = root.getElementById(newNode.getAttribute("targetId"));
+
+  forEachChild(newNode, function (newProp) {
+    var oldProp = oldNode.getElementsByTagName(newProp.tagName)[0];
+    oldNode.replaceChild(newProp, oldProp);
+  });
+}
+
+function processCreateNodes(root, createNode) {
+  forEachChild(createNode, function (child) {
+    processCreateNode(root, child);
+  });
+}
+
+function processCreateNode(root, newParent) {
+  var oldParent = root.getElementById(newParent.getAttribute("targetId"));
+
+  forEachChild(newParent, function (child) {
+    oldParent.appendChild(child);
+  });
+}
+
+function processDeleteNode(root, deleteNode) {
+  forEachChild(deleteNode, function (child) {
+    var nodeToDelete = root.getElementById(child.getAttribute("targetId"));
+    if (!nodeToDelete) return;
+
+    nodeToDelete.parentNode.removeChild(nodeToDelete);
+  });
+}
+
+function forEachChild(parent, callback) {
+  var children = parent.children.slice();
```

Comment:
Why are you calling `slice` here? It makes a shallow copy of the array, but we aren't modifying the array object at all so I don't think its needed.

Reply:
I'll have to dive into this again. It seemed like the list was changing with some iterations.

Reply:
After taking another look at this, I now see what the issue is. There is actually modification being done here. 1 example:
Suppose the `Update` element had a `Create` element with 2 placemarks.
- The loop would go through the `Create` elements children (the placemarks)
- The first placemark is appended to the original kml that is to be updated
- This removes it from the `Create` element children, so now there's only 1 left